### PR TITLE
vorbis: add 1.3.7 and components

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **vorbis/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/vorbis/all/conanfile.py [hunk 1]
```diff
@@ -10,66 +9,80 @@ class VorbisConan(ConanFile):
...
+        self.cpp_info.components["vorbisenc"].names["cmake_find_package_multi"] = "vorbisenc"
+        self.cpp_info.components["vorbisenc"].names["pkg_config"] = "vorbisenc"
+        self.cpp_info.components["vorbisenc"].libs = ["vorbisenc"]
+        self.cpp_info.components["vorbisenc"].requires = ["vorbismain"]
+        # vorbisfile
+        self.cpp_info.components["vorbisfile"].names["cmake_find_package"] = "vorbisfile"
+        self.cpp_info.components["vorbisfile"].names["cmake_find_package_multi"] = "vorbisfile"
+        self.cpp_info.components["vorbisfile"].names["pkg_config"] = "vorbisfile"
+        self.cpp_info.components["vorbisfile"].libs = ["vorbisfile"]
+        self.cpp_info.components["vorbisfile"].requires = ["vorbismain"]
```
Comment (madebr) [inline recipes/vorbis/all/conanfile.py:88]:
```suggestion
        self.cpp_info.components["vorbisfile"].requires = ["vorbismain"]

        # VorbisConfig.cmake defines components 'Enc' and 'File',
        # which create the imported targets Vorbs::vorbisenc and Vorbis::vorbisfile
        self.cpp_info.components["Enc"].requires.append("libvorbisenc")
        self.cpp_info.components["File"].requires.append("libvorbisenc")
```

File: recipes/vorbis/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -1,11 +1,11 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.1)
 project(test_package)
 
-set(CMAKE_VERBOSE_MAKEFILE TRUE)
-
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup()
 
+find_package(Vorbis REQUIRED vorbisenc CONFIG)
```
Comment (madebr) [inline recipes/vorbis/all/test_package/CMakeLists.txt:7]:
```suggestion
find_package(Vorbis REQUIRED COMPONENTS Enc CONFIG)
```

File: recipes/vorbis/all/conanfile.py (hunk 1 above)
Comment (madebr) [inline recipes/vorbis/all/conanfile.py:88]:
```suggestion
        self.cpp_info.components["vorbisfile"].requires = ["vorbismain"]

        # VorbisConfig.cmake defines components 'Enc' and 'File',
        # which create the imported targets Vorbs::vorbisenc and Vorbis::vorbisfile
        self.cpp_info.components["Enc"].requires.append("libvorbisenc")
        self.cpp_info.components["File"].requires.append("libvorbisenc")
```

File: recipes/vorbis/all/patches/link-libm.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+--- a/lib/CMakeLists.txt
++++ b/lib/CMakeLists.txt
+@@ -84,6 +84,10 @@ if (NOT BUILD_FRAMEWORK)
+     set_target_properties(vorbisfile PROPERTIES SOVERSION ${VORBISFILE_VERSION_INFO})
+ 
+     target_link_libraries(vorbis ${OGG_LIBRARIES})
++    find_library(M_LIB m)
++    if(M_LIB)
++        target_link_libraries(vorbis ${M_LIB})
++    endif()
```
Comment (madebr) [inline recipes/vorbis/all/patches/link-libm.patch:10]:
```suggestion
+    if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+        target_link_libraries(vorbis m)
+    endif()
```

File: recipes/vorbis/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/vorbis/all/conanfile.py:88]: In config file (or target file included by config file), which line creates Enc and File components as alias of vorbisenc and vorbisfile components?

Comment (madebr) [reply recipes/vorbis/all/conanfile.py:88]: See https://github.com/xiph/vorbis/blob/master/cmake/VorbisConfig.cmake.in

Also, I accidentally created a vorbis pr myself before I saw your pr:
https://github.com/conan-io/conan-center-index/pull/2605

Comment (madebr) [reply recipes/vorbis/all/conanfile.py:88]: libsndfile cmake is using these components

Comment (madebr) [reply recipes/vorbis/all/conanfile.py:88]: Also note the 2nd vorbis patch in #2605, to 100% find the ogg dependency.

Comment (SpaceIm) [reply recipes/vorbis/all/conanfile.py:88]: Still don't understand where Vorbis::Enc and Vorbis::File import libs are created (not in VorbisTargets). Some magic in check_required_components?

Comment (madebr) [reply recipes/vorbis/all/conanfile.py:88]: Vorbos::Enc and Vorbis::File are not generated.

`Enc` and `File` are the names of the components that check whether `Vorbis::vorbisenc` and `Vorbis::vorbisfile` exist.
Also, `Vorbis_Enc_FOUND` and `Vorbis_File_FOUND` should be initialized.

Comment (madebr) [reply recipes/vorbis/all/conanfile.py:88]: The name of the component != name of the target.

File: recipes/vorbis/all/patches/link-libm.patch (hunk 3 above)
Comment (SpaceIm) [reply recipes/vorbis/all/patches/link-libm.patch:10]: Ok, but I have to recreate the patch, number of lines is different.

File: recipes/vorbis/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/vorbis/all/conanfile.py:88]: understood ;)

There is a typo, it's not `libvorbisenc` 2 times, but `vorbisenc` and `vorbisfile`. Also I would suggest to give less generic conan component name for Enc and File (possible side effects like name collision for `pkg_config` generator) and explicitly set `cmake_find_package` and `cmake_find_package_multi` names.

Comment (madebr) [reply recipes/vorbis/all/conanfile.py:88]: Cool!
Yeah, using `Enc` as a component name is very generic. Something longer and settings only the cmake names would avoid potential collisions.

**Review by danimtb — APPROVED** [review]

**Pull request merged**